packages/mobile/src/in-house-liquidity/index.ts

```diff
@@ -99,73 +88,48 @@ interface SignAndFetchParams {
- * @param {params.walletAddress} walletAddress
- * @param {params.requestOptions} RequestInit all the normal fetch options
+ * @param {params.accountMTWAddress} accountMTWAddress
+ * @param {params.requestOptions} requestOptions all the normal fetch options
  * @returns {Response} response object from the fetch call
  */
 export const signAndFetch = async ({
   path,
   accountMTWAddress,
-  walletAddress,
+  dekPrivate,
   requestOptions,
 }: SignAndFetchParams): Promise<Response> => {
-  const date = new Date()
-  const authAndDateHeaders = await getAuthAndDateHeaders({
-    httpVerb: requestOptions.method,
-    requestPath: path,
-    date,
-    accountMTWAddress,
-    walletAddress,
-    requestBody: requestOptions.body,
-  })
+  const authHeader = await getAuthHeader({ accountMTWAddress, dekPrivate })
   return fetch(`${networkConfig.inHouseLiquidityURL}${path}`, {
     ...requestOptions,
     headers: {
       ...requestOptions.headers,
-      ...authAndDateHeaders,
+      Authorization: authHeader,
     },
   })
 }
 
-interface GetAuthAndDateHeadersParams {
-  httpVerb: string | undefined
-  requestPath: string
-  date: Date
-  accountMTWAddress: string
-  walletAddress: string
-  requestBody?: BodyInit | null
-}
 /**
- * Gets the auth and date headers that IHL expects as a signature on most requests
- *
- * The behavior is slightly different between GET requests and other types of requests
+ * Gets the auth header that IHL expects as a signature on most requests
  *
- * @param {params.httpVerb} string GET, POST
- * @param {params.requestPath} string like /persona/get/foo
- * @param {params.date} Date date object
- * @param {params.accountMTWAddress} accountAddress
- * @param {params.walletAddress} walletAddress
- * @param {params.requestBody} string optional request body
- * @returns {{Date, Authorization}} date and authorization headers
+ * @param {params.accountMTWAddress} accountMTWAddress
+ * @param {params.dekPrivate} dekPrivate : private data encryption key
+ * @returns authorization header
  */
-export const getAuthAndDateHeaders = async ({
-  httpVerb = '',
-  requestPath,
-  date,
+export const getAuthHeader = async ({
   accountMTWAddress,
-  walletAddress,
-  requestBody,
-}: GetAuthAndDateHeadersParams): Promise<{ Date: string; Authorization: string }> => {
-  const dateString = date.toUTCString()
-  const message =
-    httpVerb.toLowerCase() === 'get'
-      ? `${httpVerb.toLowerCase()} ${requestPath} ${dateString}`
-      : `${httpVerb.toLowerCase()} ${requestPath} ${dateString} ${requestBody}`
-  const serializedSignature = await getSerializedSignature({ message, accountMTWAddress })
-  const authorization = `Valora ${walletAddress}:${serializedSignature}`
-  return {
-    Date: dateString,
-    Authorization: authorization,
-  }
+  dekPrivate,
+}: {
+  accountMTWAddress: string
+  dekPrivate: string
+}): Promise<string> => {
+  const keyEncoder = new KeyEncoder('secp256k1')
```

Comment:
Does this need to be done on every request or can it instead be done once at the top of the file?

Reply:
Nothing in the docs about this. I looked at the code for encodePublic and encodePrivate and neither appears to mutate the keyEncoder. (Even if they did, the private/public DEKs for the user should not change.) So I think it's fine to have it at the top.